### ISSUE TITLE
Files sometime not refreshing until restarting editor.

### ISSUE THREAD

**Issue by avencherus** [godotengine/godot#21494]
Using a 64-bit Windows build of Godot 3.1 from a few hours ago.

So far this was a one off event.  I'm not exactly sure how to reproduce this one yet.  I'll post back if I have an success or re-encounter this.

Was testing out another issue, and saved some changes in an external editor.  I removed the comment from line 2 as seen below.

![sublime_text_2018-08-27_20-03-07](https://user-images.githubusercontent.com/13004169/44676914-398bf280-aa3d-11e8-8f23-625fdaf220df.png)

When I returned to the editor the old code was persisting, even when closing and reopening the file.

![godot master_2018-08-27_20-02-44](https://user-images.githubusercontent.com/13004169/44676957-56c0c100-aa3d-11e8-90bb-d3ea74fe00e3.png)

I ran it as it was, and the ground truth was that is was actually edited, because this line was going to cause an error and did, because I used the same class name in another script.

![godot master_2018-08-27_20-02-50](https://user-images.githubusercontent.com/13004169/44677003-7d7ef780-aa3d-11e8-9a5b-a612d7325d34.png)

I went back to the script after running it, and it was still commented out as if it were the old version.

It only resolved by a restart of the editor.

I was also getting some errors like these a little bit earlier, and am including them just in case they're related if there is something that can go amiss with the file reading/locking.

**Note**:  The only png in this project is the default icon.png.

![godot master_2018-08-27_19-57-17](https://user-images.githubusercontent.com/13004169/44677107-cc2c9180-aa3d-11e8-90bb-8ea99fe6a9eb.png)

**Comment by avencherus** [comment]
Still periodically happening as of 1b62b2c.

At the moment I have to copy and paste my outside work into the script editor.  The project has to be closed and re-opened, and the soft reload option in the script editor doesn't work.

**Comment by avencherus** [comment]
Closing.  I believe this to be resolved by: https://github.com/godotengine/godot/pull/22069

Related to: https://github.com/godotengine/godot/issues/21956

Currently, there is another issue with the editor and tool script not reloading external code changes: https://github.com/godotengine/godot/issues/22144

**Comment by chanon** [comment]
> the old code was persisting, even when closing and reopening the file.

I am seeing this happening too. Difference is that I wasn't editing in an external editor. 
I was on commit: 94403af7ad9d853227fac72241953c4a0c2a4aa1
Which was after #22069 was committed.

I made changes to a .gd script, committed and pushed them in git, worked on some other stuff, and then came back to the .gd script and it was the old version before I made the changes. I closed the file in the script editor and re-opened it and it was still the old version. I opened the file on disk in another program and it was the edited committed version. So Godot was showing something completely different from what was on disk, as if it was sticking to its cached version.

**Comment by chanon** [comment]
And it happened a second time in a different file, where it hit a bug in my code, but the debugger stopped on the wrong line. After looking at it I realized that it was again the wrong version of the script file (not the one on disk) which is why the debugger stopped on the wrong line ... the line with the bug wasn't in the version the editor was displaying.

**Comment by avencherus** [comment]
Sure thing, I'll reopen it.

**Comment by avencherus** [comment]
Seems to update now.  The only thing I get is an error in the console.  Would this be something you want to reclassify?

![image](https://user-images.githubusercontent.com/13004169/51645014-f9dd1e80-1f7a-11e9-9c24-7349ff352bc3.png)


It might be a new issue, because I notice that this stops the syntax checking until you change something in the script and save it from inside the editor.

![image](https://user-images.githubusercontent.com/13004169/51645132-8556af80-1f7b-11e9-8779-1ef2c1a3d2d1.png)

After adding a space on the end and saving it.

![image](https://user-images.githubusercontent.com/13004169/51645156-a1f2e780-1f7b-11e9-9d3b-7be8004a6989.png)

**Comment by nicolasdiehl** [comment]
Godot 3.1.beta.calinou.9ca6ffa:
open VehicleBody in godot,
open VehicleBody in text editor,
edit mass in text editor and save,
close VehicleBody in godot without saving,
open VehicleBody in godot
--> text editor complains that mass of VehicleBody changed.

**Comment by akien-mga** [comment]
Seems like a duplicate of #20250 (at least the comment for which this issue was reopened, once the original one had been closed).

**Comment by madamc** [comment]
I'm experiencing this same issue. I can make no new changes to my code. Restarting the editor and Visual Studio Code does nothing. I'm using Godot Engine v3.2.1.stable.mono.official.

I'm not sure if this is the cause, but I also receive these errors every second or so while the game is running:
`E 0:00:10.020   body_test_motion: Condition "!body->get_space()" is true. Returned: false
  <C++ Source>  servers/physics_2d/physics_2d_server_sw.cpp:1051 @ body_test_motion()
  <Stack Trace> :0 @ void Godot.NativeCalls.godot_icall_6_501(IntPtr , IntPtr , Godot.Vector2& , Godot.Vector2& , Boolean , Int32 , System.Single& , Boolean , Godot.Vector2& )()
                KinematicBody2D.cs:96 @ Godot.Vector2 Godot.KinematicBody2D.MoveAndSlide(Godot.Vector2 , System.Nullable`1[Godot.Vector2] , Boolean , Int32 , Single , Boolean )()
                BulletOrb.cs:77 @ void BulletOrb._PhysicsProcess(Single )()
`
`E 0:00:10.024   body_test_ray_separation: Condition "!body->get_space()" is true. Returned: false
  <C++ Source>  servers/physics_2d/physics_2d_server_sw.cpp:1063 @ body_test_ray_separation()
  <Stack Trace> :0 @ void Godot.NativeCalls.godot_icall_6_501(IntPtr , IntPtr , Godot.Vector2& , Godot.Vector2& , Boolean , Int32 , System.Single& , Boolean , Godot.Vector2& )()
                KinematicBody2D.cs:96 @ Godot.Vector2 Godot.KinematicBody2D.MoveAndSlide(Godot.Vector2 , System.Nullable`1[Godot.Vector2] , Boolean , Int32 , Single , Boolean )()
                BulletOrb.cs:77 @ void BulletOrb._PhysicsProcess(Single )()
`
